portal-impl/src/com/liferay/portal/cache/key/CacheKeyGeneratorUtil.java

```diff
@@ -19,45 +16,41 @@
  */
 public class CacheKeyGeneratorUtil {
 
-	public static CacheKeyGenerator getCacheKeyGenerator(String cacheName) {
-		ThreadLocalCache<CacheKeyGenerator> threadLocalCacheKeyGenerators =
-			ThreadLocalCacheManager.getThreadLocalCache(
-				Lifecycle.ETERNAL, CacheKeyGeneratorUtil.class.getName());
+	public static final String DEFAULT = "DEFAULT";
+
+	public static final String HASH_CODE_HEX_STRING = "HASH_CODE";
+
+	public static final String MESSAGE_DIGEST = "MESSAGE_DIGEST";
+
+	public static final String SIMPLE = "SIMPLE";
 
-		CacheKeyGenerator cacheKeyGenerator = threadLocalCacheKeyGenerators.get(
+	public static CacheKeyGenerator getCacheKeyGenerator(String cacheName) {
+		CacheKeyGenerator cacheKeyGenerator = _cacheKeyGeneratorsMap.get(
```

Comment:
@julschong After your change the parameter passed is no longer a "cacheName", please rename the variable.